### DISCUSSION TITLE
Nano 33 BLE opendeck 3.0.1 not working & not showing up in configurator

### DISCUSSION THREAD

**Discussion by eurorackworld** [shanteacontrols/OpenDeck#182] [Q&A] [answered]
I had just one potentiometer connected to test some stuff and I was browsing through the online configurator. Then I choose 'factory reset' , but after submitting and closing the info box about the reset it looks like the board isn't doing anything (anymore). Now when I just connect the board via an USB cable , there is a led turning on at the Nano board and the red power led on the opendeck board. But nothing is happening anymore on the board, and the configurator is not showing the board anymore.

Is there anything I can do to get this to work again?

p.s. If I power up the board now the RGB led in the middle of the Nano blinks 3 times and then the led on the right of the usb connector turns on (the L led? Built in led?)

**Comment by paradajz** [comment, marked as answer]
Right below the indicator LEDs, there is a BTLDR connector. If you short circuit those two pins together and then power on the board, does the configurator show firmware update screen?

**Reply by eurorackworld** [reply]
That works fine, I'm able to upload new firmware. I tried a few but I didn't get it all to work again, but I'm not sure if I use the right firmware. I used the ones here: https://github.com/shanteacontrols/OpenDeck/releases 

But not sure if to use one of the opendeck versions or the nano ble as I have a nano ble device on the opendeck pcb

**Reply by paradajz** [reply]
You need to use opendeck3a.sysex file. Did you try that?

**Reply by eurorackworld** [reply]
Thanks that worked. The board is now being found again, although I do get into that same bootloader mode again which I also saw when I shortened the BTLDR pins before., now it shows this without BTLDR being shortenend. I remember I has such a thing earlier, but no idea anymore how I got out of this bootloader mode? (as this way the board of course isn't useable yet)

**Reply by paradajz** [reply]
Ehm, not sure I understand. What exactly worked then? You've entered the bootloader mode, you've flashed opendeck3a.sysex firmware, and then what? After it finished it goes back to bootloader mode?

**Reply by eurorackworld** [reply]
The initial situation was that I didn't see any response in the online configurator when plugging in the board. Then I shortened the BTLDR pins and when powering up now the online configurator showed up showing the firmware update screen. I've update it with the suggested file and now I remove the shorten between de BTLDR pins and when I power up the board now, the online configurator shows, but is showing the same firmware update screen. 

So the progress was that at least the online configurator shows contact with the board without having to short the pins:)

But now if I power up the device it does  only show the firmware update screen and I'm stuck in that screen now.

**Reply by eurorackworld** [reply]
P.s.  I had such a thing before and I managed to get out but no idea how anymore...

**Reply by eurorackworld** [reply]
I see now that after the firmware update, the led in the middle of the nano is blinking. But when powering off and power up again is shows still the firmware update menu.

**Reply by paradajz** [reply]
Do you remember which firmware you had originally on the board when you've received it?

**Reply by eurorackworld** [reply]
No, don't know. The back says "opendeck board revision: 3.0.1. 2022.05.  And I received the board around June 15th 2022. Not sure if you normally put the latest firmware on it before shipping.

**Reply by paradajz** [reply]
Not sure what's going on. I've tried to flash my board with the firmware released around that time, then I've tried to update it through the browser to the latest version and everything is working. Since there aren't really any other options you could try, I can send you another pre-flashed Arduino Nano 33 BLE with the latest firmware.

**Reply by eurorackworld** [reply]
Hey thanks voor checking and testing. Thanks for the offer as well for the pre flashed nano, but I know for sure that I had something like this earlier when I tried a firmware upgrade (somewhere when I just had the board), then it got stuck into bootloader mode as well and I somehow managed to get it to work again..I'll try some different firmwares.

Just to be sure, the other files (bin and another one) I did try to upload as well (before knowing I had to use the sysex, although I saw it was clearly mentioned...), But I assume those files will do no harm? I tried again to see what happened and they just give errors except on the wait request :p.

So I'll try to fix it myself, maybe I'm somehow able to follow what I did last time. I'll let you know the result.

**Reply by eurorackworld** [reply]
Tried various things (earliest firmware, both 3a and 3b versions, most recent firmware)

When I update the firmware using the latest release, then de nano turns into some kind of bootloop (3 flashes of the main rgb led, then the led next to the usb turns on, after few seconds the whole sequence repeats). 

I'll contact you via email further. Thanks for thinking along in here.

**Reply by paradajz** [reply]
You cannot flash anything other than opendeck3a.sysex for that board and expect things to work, especially if you tried to flash firmware for completely different microcontroller (3b uses STM32, 3a is nRF52).
